main.go

```diff
@@ -135,6 +135,25 @@ func handlePprof(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
 	}
 }
 
+func computeRoutePrefix(prefix string, externalURL *url.URL) string {
```

Comment:
The inline comments in this function are essentially just describing the (very simple) code and thus make it actually less readable. I guess the point you are trying to make is that this code implements the spec given by the flags. How about turning them into a doc comment before the function:

```Go
// computeRoutePrefix returns the effective route prefix based on the provided
// flag values for --web.route-prefix and--web.external-url. With prefix empty,
// the path of externalURL is used instead. A prefix "/" results in an empty
// returned prefix. Any non-empty prefix is normalized to start, but not to end,
// with "/".
```